functions/Find-DbaOrphanedFile.ps1

```diff
@@ -0,0 +1,225 @@
+            $method = 'remote'
+        }
+
+		#Write-Output "Attempting to connect to SQL Server.."
+		
+		$sourceserver = Connect-SqlServer -SqlServer $sqlserver -SqlCredential $SqlCredential
+
+        # Get all the database files
+        $databasefiles = Get-SqlFileStructure -server $sourceserver
+
+        # Check if there are any database files retrieved
+        if($databasefiles.count -ge 1)
+        {
+            # Create the orphaned files variable
+            $orphanedfiles = @() 
+
+            # Getthe default data and log directories from the instance
+            $Path += $sourceserver.RootDirectory + "\DATA"
+            
+
+            # Get the default data and log paths
+            $Path += Get-SqlDefaultPaths $server data
+            $Path += Get-SqlDefaultPaths $server log
+
+            # Add the system database paths if not already included
+            if($Path -notcontains $sourceserver.MasterDBPath)
+            {
+                $Path += $sourceserver.MasterDBPath
+            }
+            if($Path -notcontains $sourceserver.MasterDBLogPath)
+            {
+                $Path += $sourceserver.MasterDBLogPath
+            }
+
+            # Create the array to hold the files on disk
+            $diskfiles = @()
+
+            # Loop through each of the directories and get all the data and log file related files
+            foreach($directory in $Path)
+            {
+                # Cleanup directory
+                if($directory.EndsWith("\")) 
+                {
+                    $directory = $directory.TrimEnd("\")
+                }
+                
+                # Check the method and change to UNC admin paths
+                if($method -eq 'remote')
+                {
+                    # Set the directory to administrative UNC paths
+                    $directory = Join-AdminUnc -servername $servername -FilePath $directory
+
+                }
+                
+                # Check if the path exists
+                if(Test-Path $directory)
+                {
+                    # Get the files on disk
+                    $diskfiles += Get-ChildItem -Path "$directory\*.*" -Include *.mdf,*.ldf,*.ndf 
+                }
+                
+            }
+
+            # If there are files found
+            if($diskfiles.count -ge 1)
+            {
+                switch($method)
+                {
+                    'local' { $dbfiles = $databasefiles.Tables[0].filename }
+                    'remote' { $dbfiles = $databasefiles.Tables[0].filename | Where-Object {$_ -match ":"} | ForEach-Object {$_ -replace ':','$'} | ForEach-Object {"\\$servername\" + $_} }
+                }
+
+                # Compare the two lists and save the items that are not in the database file list 
+                $orphanedfiles = (Compare-Object -ReferenceObject ($dbfiles) -DifferenceObject $diskfiles.FullName).InputObject
+            }
+                
+            return $orphanedfiles

```

Comment:
Works! But please return two columns LocalPath and RemotePath, filled with local (M:\etc) and remote (\\server\m$\etc)